hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java

```diff
@@ -150,6 +155,77 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
           .addCacheEntry(new CacheKey<>(key),
             new CacheValue<>(Optional.of(snapshotInfo), transactionLogIndex));
 
+      // Create the snapshot checkpoint
+      OmSnapshotManager.createOmSnapshotCheckpoint(omMetadataManager,
+          snapshotInfo);
```

Comment:
I think OmSnapshotManager.createOmSnapshotCheckpoint() has to be called from within OMSnapshotCreateResponse:addToDBBatch().  The special double buffer changes @hemantk-12 made depend on that.

Reply:
You are right. I'm going to move the checkpointing back to `Response#addToDBBatch`

Reply:
The checkpointing and the deleteRange call both should be in addToDBBatch() right?

Reply:
@GeorgeJahad Correct. I've adjusted the logic. The deletedTable clean up is now done in `createOmSnapshotCheckpoint` as well.

Reply:
We had discussed this in our internal snapshot syncs. Lets not change the place where we issue the snapshot create. We have placed the checkpoint creation such that it will not cause the OM nodes to diverge from each other.